### PR TITLE
#1966 - Fix sonarcloud critical and major issues (part 1)

### PR DESCRIPTION
**Pull request by sh16011993**
This PR addresses the following sonarcloud critical issues:

- Change `var` to use `const` or `let`.
- Provide a comparator as an argument to the sort function.

### CODE REVIEW

File: sources/packages/backend/workflow/BPMN/assessment-gateway/extract-dynamic-data-exceptions.js
Comment (andrewsignori-aot) [inline sources/packages/backend/workflow/BPMN/assessment-gateway/extract-dynamic-data-exceptions.js:1]: The javascript files inside the BPMN are legacy javascript files and they must be ES5 compliant so they should not be changed. Please revert these changes.

File: sources/packages/backend/workflow/scripts/api-response-parser.js
Comment (andrewsignori-aot) [inline sources/packages/backend/workflow/scripts/api-response-parser.js:1]: Same from https://github.com/bcgov/SIMS/pull/1969/files#r1206160743

File: sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts [hunk 1]
```diff
@@ -81,7 +81,13 @@ export abstract class SFTPIntegrationBase<DownloadType> {
 
     return filesToProcess
       .map((file) => path.join(remoteDownloadFolder, file.name))
-      .sort();
+      .sort((file1: string, file2: string) => {
```
Comment (andrewsignori-aot) [inline sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts:84]: It can be simplified and still be Sonarcloud compatible, IMO.
```ts
.map((file) => path.join(remoteDownloadFolder, file.name))
.sort((a, b) => a.localeCompare(b));
```

**Review by andrewsignori-aot — COMMENTED** [review]
Thanks for looking into Sonarcloud issues. Please take a look at the comments.

**Comment by andrewsignori-aot** [comment]
> This PR addresses the following sonarcloud critical issues:
> 
> * Change `var` to use `const` or `let`.
> * Provide a comparator as an argument to the sort function.
> 
> The following critical sonar issue has not been addressed as a part of this PR:
> 
> * Refactor function to reduce its Cognitive Complexity from 19 to the 15 allowed.

Is the last item in the description part of this PR? If not, please remove it.

File: sources/packages/backend/workflow/BPMN/assessment-gateway/extract-dynamic-data-exceptions.js
Comment (guru-aot) [reply sources/packages/backend/workflow/BPMN/assessment-gateway/extract-dynamic-data-exceptions.js:1]: As mentioned by @andrewsignori-aot , these files created as a part of a helper js in bpmn should not be changed. Please revert, thanks

**Comment by dheepak-aot** [comment]
Thanks for taking care of Sonar code quality issues. Please revert the changes on workflow/*.js files as mentioned by other devs.

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @sh16011993

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, looks good 👍

**Review by andrepestana-aot — APPROVED** [review]
Looks good!
